src/libsystemd/sd-journal/journal-internal.h

```diff
@@ -151,3 +152,15 @@ static inline bool JOURNAL_ERRNO_IS_UNAVAILABLE_FIELD(int r) {
                       E2BIG,            /* Field too large for pointer width */
                       EPROTONOSUPPORT); /* Unsupported compression */
 }
+
+#define journal_stream_path(log_namespace)                                              \
```

Comment:
Nah, please implement it as a normal function with strjoin().

Reply:
Well, that way we'd lose the "async signal safe" attribute. I'm not sure if that's acceptable?

Reply:
I'm not too concerned that people would actually call these in signal handlers, though...

Reply:
Hmm, so obviously people do rely on this: https://github.com/systemd/systemd/issues/7912.

Reply:
Hmm, then please add comment why this is a macro.